Subject: [PATCH] gitweb: Do not use application/xhtml+xml content type

Using that content type seems to give us no benefits and it makes many
browsers (e.g. Firefox) exceedingly strict, which is troublesome
for example when showing user-supplied README.html files.

Signed-off-by: Petr Baudis <pasky@suse.cz>
---
gitweb/gitweb.perl | 16 +++++-----------
1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/gitweb/gitweb.perl b/gitweb/gitweb.perl
index a8009d8d..ef7f2a69 100755
--- a/gitweb/gitweb.perl
+++ b/gitweb/gitweb.perl
@@ -4071,17 +4071,11 @@ sub get_page_title {
 }
 
 sub get_content_type_html {
-	# require explicit support from the UA if we are to send the page as
-	# 'application/xhtml+xml', otherwise send it as plain old 'text/html'.
-	# we have to do this because MSIE sometimes globs '*/*', pretending to
-	# support xhtml+xml but choking when it gets what it asked for.
-	if (defined $cgi->http('HTTP_ACCEPT') &&
-	    $cgi->http('HTTP_ACCEPT') =~ m/(,|;|\s|^)application\/xhtml\+xml(,|;|\s|$)/ &&
-	    $cgi->Accept('application/xhtml+xml') != 0) {
-		return 'application/xhtml+xml';
-	} else {
-		return 'text/html';
-	}
+	# We do not ever emit application/xhtml+xml since that gives us
+	# no benefits and it makes many browsers (e.g. Firefox) exceedingly
+	# strict, which is troublesome for example when showing user-supplied
+	# README.html files.
+	return 'text/html';
 }
 
 sub print_feed_meta {
---
